tests/unit/frameworks/kfp/constructs/cloudrun_test.py

```diff
@@ -0,0 +1,425 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for cloudrun constructs kfp module."""
+
+# pylint: disable=C0103
+# pylint: disable=line-too-long
+# pylint: disable=missing-function-docstring
+# pylint: disable=protected-access
+
+import pytest
+from AutoMLOps.utils.utils import write_yaml_file
+from AutoMLOps.utils.constants import (
+    GENERATED_LICENSE,
+    LEFT_BRACKET,
+    RIGHT_BRACKET
+)
+from AutoMLOps.frameworks.kfp.constructs.cloudrun import KfpCloudRun
+
+# Create defaults file contents to test
+DEFAULTS1 = {
+    'gcp':
+        {
+            'af_registry_location': 'us-central1',
+            'af_registry_name': 'vertex-mlops-af',
+            'base_image': 'python:3.9-slim',
+            'cb_trigger_location': 'us-central1',
+            'cb_trigger_name': 'automlops-trigger',
+            'cloud_run_location': 'us-central1',
+            'cloud_run_name': 'run-pipeline',
+            'cloud_tasks_queue_location': 'us-central1',
+            'cloud_tasks_queue_name': 'queueing-svc',
+            'cloud_schedule_location': 'us-central1',
+            'cloud_schedule_name': 'AutoMLOps-schedule',
+            'cloud_schedule_pattern': '0 */12 * * *',
+            'cloud_source_repository': 'AutoMLOps-repo',
+            'cloud_source_repository_branch': 'automlops',
+            'gs_bucket_name': 'automlops-sandbox-bucket',
+            'pipeline_runner_service_account': 'vertex-pipelines@automlops-sandbox.iam.gserviceaccount.com',
+            'project_id': 'automlops-sandbox',
+            'vpc_connector': 'No VPC Specified'
+        }
+    }
+DEFAULTS2 = {
+    'gcp':
+        {
+            'af_registry_location': 'us-central1',
+            'af_registry_name': 'vertex-mlops-af',
+            'base_image': 'python:3.9-slim',
+            'cb_trigger_location': 'us-central1',
+            'cb_trigger_name': 'automlops-trigger',
+            'cloud_run_location': 'us-central1',
+            'cloud_run_name': 'run-pipeline',
+            'cloud_tasks_queue_location': 'us-central1',
+            'cloud_tasks_queue_name': 'queueing-svc',
+            'cloud_schedule_location': 'us-central1',
+            'cloud_schedule_name': 'AutoMLOps-schedule',
+            'cloud_schedule_pattern': '0 */12 * * *',
+            'cloud_source_repository': 'AutoMLOps-repo',
+            'cloud_source_repository_branch': 'automlops',
+            'gs_bucket_name': 'automlops-sandbox-bucket',
+            'pipeline_runner_service_account': 'vertex-pipelines@automlops-sandbox.iam.gserviceaccount.com',
+            'project_id': 'automlops-sandbox',
+            'vpc_connector': 'No VPC Specified'
+        }
+    }
+
+@pytest.fixture(name='defaults_dict', params=[DEFAULTS1, DEFAULTS2])
+def fixture_defaults_dict(request, tmpdir):
```

Comment:
Not sure what types to add here since the fixture_defaults_dict function never gets called, the test just pulls from it as needed when defaults_dict is passed

Reply:
Hey @Akhan221 I forwarded you an email regarding it but @srastatter was looking for something along these lines https://stackoverflow.com/questions/65334215/how-can-request-param-be-annotated-in-indirect-parametrization